p2p/session.go

```diff
@@ -213,6 +213,9 @@ func (s *session[H]) doRequest(
 			"err", err,
 			"peer", stat.peerID,
 		)
+		if errors.Is(err, header.ErrNotFound) {
```

Comment:
This needs a comment with an explanation.

Reply:
Should that be combined with the switch above?

Reply:
no, it should go after with, bc we need to send a failed request to the channel and add peer to the queue to decrease the chance of the range being requested from the same peer multiple times in a row.